### ISSUE TITLE
Schema enforcement: m3db client config requires schema deploy ID

### ISSUE THREAD

**Issue by Haijuncao** [m3db/m3#1700]
Currently, m3db client loads schema history from configuration files bundled with binary.  The schema is supposed to have been deployed to m3db meta node (etcd), but there is no enforcement.

This tickets covers the enforcement on the client side, we have another ticket that covers server side enforcement. This ticket should be done first before the server side enforcement.

As a first step, we should require schema deploy ID to be set as part of m3db client configuration, and schema deploy ID should be validated by checking it against meta node (etcd). The best way to validate it is to download the schema history from etcd and check if the deploy ID is present (may not be the latest). 

Since schema history needed to be downloaded anyway, we might as well use it to populate in memory schema registry, instead of what we do currently: from configuration.
